ambry-utils/src/main/java/com/github/ambry/utils/Utils.java

```diff
@@ -1499,6 +1499,17 @@ public static int compareTimestamp(Timestamp timestamp, long otherTimeMs) {
     return Utils.compareTimes(timestampToMs(timestamp), otherTimeMs);
   }
 
+  /**
+   * Get the latest timestamp, and if one of the timestamp is null, return the non-null timestamp.
+   */
+  public static Timestamp getLatestTimeStamp(Timestamp timestamp1, Timestamp timestamp2) {
```

Comment:
when timestamp1 is not null but timestamp2 is null, this function will return the null one.
It's because compareTimestamp thinks -1 is infinity and it's bigger. 

Need a fix for it.